azure/template.json

```diff
@@ -13,75 +37,118 @@
       "type": "string",
       "defaultValue": "Basic"
     },
-    "mySqlSkuFamily": {
-      "type": "string",
-      "defaultValue": "Gen5"
-    },
     "mySqlStorageMB": {
       "type": "string",
       "defaultValue": "51200"
-    },
-    "mySqlAdminLogin": {
-      "type": "string"
-    },
-    "mySqlAdminPassword": {
-      "type": "securestring"
     }
   },
   "variables": {
     "deploymentUrlBase": "https://raw.githubusercontent.com/SkillsFundingAgency/das-platform-building-blocks/master/templates/",
-    "mySqlServerName": "[concat('das-', parameters('resourceEnvironmentName'), '-graf-msql')]",
-    "storageAccountName": "[toLower(concat('das', parameters('resourceEnvironmentName'), 'grafstr'))]"
+    "resourceNamePrefix": "[toLower(concat('das-', parameters('resourceEnvironmentName'), '-graf'))]",
+    "mySqlServerName": "[concat(variables('resourceNamePrefix'), '-msql')]",
+    "resourceGroupName": "[concat(variables('resourceNamePrefix'),'-rg')]",
+    "storageAccountName": "[toLower(concat(replace(variables('resourceNamePrefix'), '-', ''), 'str'))]"
   },
   "resources": [
     {
-      "name": "[variables('mySqlServerName')]",
-      "type": "Microsoft.DBforMySQL/servers",
-      "apiVersion": "2017-12-01",
-      "location": "[resourceGroup().location]",
-      "sku": {
-        "name": "[parameters('mySqlSkuName')]",
-        "tier": "[parameters('mySqlSkuTier')]",
-        "family": "[parameters('mySqlSkuFamily')]"
-      },
+      "apiVersion": "2020-06-01",
+      "name": "[variables('resourceGroupName')]",
+      "type": "Microsoft.Resources/resourceGroups",
+      "location": "[parameters('resourceGroupLocation')]",
+      "tags": "[parameters('tags')]",
       "properties": {
-        "administratorLogin": "[parameters('mySqlAdminLogin')]",
-        "administratorLoginPassword": "[parameters('mySqlAdminPassword')]",
-        "version": "5.7",
-        "sslEnforcement": "Enabled",
-        "storageProfile": {
-          "backupRetentionDays": 7,
-          "storageMB": "[parameters('mySqlStorageMB')]"
-        },
-        "createMode": "Default"
-      },
-      "resources": [
-        {
-          "name": "grafana",
-          "type": "databases",
-          "apiVersion": "2017-12-01",
-          "properties": {},
-          "dependsOn": [
-            "[variables('mySqlServerName')]"
+      }
+    },
+    {
+      "apiVersion": "2020-06-01",
+      "name": "mysql-database-server",
+      "resourceGroup": "[variables('resourceGroupName')]",
+      "type": "Microsoft.Resources/deployments",
+      "properties": {
+        "mode": "Incremental",
+        "template": {
```

Comment:
Should we make and use a building block instead?

Reply:
In general I would say yes but in this specific case I decided against it because MySQL databases are a type of resource that we don't want to be using unless there's no other option.